CRM/Contribute/BAO/Contribution.php

```diff
@@ -368,8 +368,8 @@ public static function calculateMissingAmountParams(&$params, $contributionID) {
             'id' => $contributionID,
             'return' => ['total_amount', 'net_amount', 'fee_amount'],
           ]);
-          $totalAmount = (isset($params['total_amount']) ? (float) $params['total_amount'] : (float) CRM_Utils_Array::value('total_amount', $contribution));
-          $feeAmount = (isset($params['fee_amount']) ? (float) $params['fee_amount'] : (float) CRM_Utils_Array::value('fee_amount', $contribution));
+          $totalAmount = (isset($params['total_amount']) ? (float) $params['total_amount'] : (float) $contribution['total_amount']);
+          $feeAmount = (isset($params['fee_amount']) ? (float) $params['fee_amount'] : (float) $contribution['fee_amount']);
```

Comment:
You'd need the `??` on these at the end to avoid notices if $contribution['...'] doesn't exist. But note the (float) would get evaluated first so you'd need brackets here too.